### PR TITLE
pull 007pig changes we lost

### PR DESCRIPTION
**Pull request by ArneBab**
We should either merge this (and get it fixed if needed) or take the explicit decision not to merge it.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,7 @@
 # Created by .ignore support plugin (hsz.mobi)
+
+src/main/java/plugins/UPnP2/ClingTest.java
```
Comment (desyncr) [inline .gitignore:3]: Why this test is ignored?

**Comment by desyncr** [comment]
Changes looks good. Although I haven't built and load the plugin locally yet. There's a little conflict on `src/main/java/plugins/UPnP2/UPnP2.java` that can be resolved easily.

File: src/main/java/plugins/UPnP2/IGDRegistryListener.java [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+import freenet.support.LogThresholdCallback;
+import freenet.support.Logger;
+import plugins.UPnP2.actions.GetSpecificPortMappingEntry;
+
+/**
+ * Cling UpnpService registry listener. Extends PortMappingListener to reuse its port mapping code.
+ */
+public class IGDRegistryListener extends PortMappingListener {
+
+    private static volatile boolean logMINOR;
```
Comment (Bombe) [inline src/main/java/plugins/UPnP2/IGDRegistryListener.java:52]: Can we please stop pretending our computers didn’t get any faster in the last 20 years and finally get rid of these extremely noisy workarounds to problems we don’t have anymore? Thank you.

File: src/main/java/plugins/UPnP2/IGDRegistryListener.java [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+        this.serviceManager = serviceManager;
+        upnpService = serviceManager.getUpnpService();
+    }
+
+    @Override
+    synchronized public void deviceAdded(Registry registry, Device device) {
+
+        Logger.normal(this, "Remote device available: " + device.getDisplayString());
+
+        Service commonService;
```
Comment (Bombe) [inline src/main/java/plugins/UPnP2/IGDRegistryListener.java:78]:
```suggestion
        Service commonService = discoverCommonService(device);
        if (commonService == null) {
            return;
        }
```

File: src/main/java/plugins/UPnP2/IGDRegistryListener.java [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+        if ((commonService = discoverCommonService(device)) == null) return;
+
+        serviceManager.addCommonService(commonService);
+
+        Service connectionService;
```
Comment (Bombe) [inline src/main/java/plugins/UPnP2/IGDRegistryListener.java:83]: Same here, please. There is no reason to split declaration and initialization, only to initialize the variable in an `if` statement (which should be a punishable offence on its own 😄).

File: src/main/java/plugins/UPnP2/IGDSubscriptionCallback.java [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+    }
+
+    @Override
+    public void eventReceived(GENASubscription sub) {
+
+        // Once we received an event, it means Cling has found at least one device. So we can set
+        // booted to true.
+        serviceManager.setBooted(true);
+
+        Map values = sub.getCurrentValues();
```
Comment (Bombe) [inline src/main/java/plugins/UPnP2/IGDSubscriptionCallback.java:114]: If the API returns a raw `Map`, please at least use `Map<?, ?>` as its type.

File: src/main/java/plugins/UPnP2/IGDRegistryListener.java (hunk 2 above)
Comment (desyncr) [reply src/main/java/plugins/UPnP2/IGDRegistryListener.java:52]: What do you mean by that? Remove that logminor thing?

Comment (Bombe) [reply src/main/java/plugins/UPnP2/IGDRegistryListener.java:52]: Yes, please. That particular pattern was introduced in the last millenium when machines generally were slower, and if you logged a lot at DEBUG but DEBUG was disabled, it incurred a noticable slowdown. Nowadays machines are fast enough that the prevented slowdown is not worth littering the source code with this kind of boilerplate.

Oh, also: *IT’S NOT EVEN USED IN THIS CLASS.*

**Comment by ArneBab** [comment]
@desyncr are working on getting this merge-ready?

**Comment by desyncr** [comment]
@ArneBab Yeah, this PR's work is over https://github.com/freenet/plugin-UPnP2/pull/3. Conversation can be taken over there.

**Comment by Bombe** [comment]
Yeah, closing this in favour of #3.

**Pull request closed without merging**